src/Squirrel/UpdateManager.ApplyReleases.cs

```diff
@@ -35,7 +35,20 @@ public ApplyReleasesImpl(string rootAppDirectory)
 
                 progress(0);
                 status("Applying updates");
-                
+
+                // Try to remove older versions before applying releases
+                try
+                {
+                    var currentVersion = updateInfo.CurrentlyInstalledVersion != null ?
```

Comment:
There is a null conditional operator in the c sharp. This line could be shortened just to:

`var currentVersion = updateInfo.CurrentlyInstalledVersion?.Version;`

or maybe even better?

`await removeOldVersionFolders(new SemanticVersion[] { updateInfo.CurrentlyInstalledVersion?.Version});`

Anyway, there is nothing bad on your implementation. Feel free to leave it as it is if you want.

Reply:
Thanks, it's good to know 😄